src/main/java/org/opensearch/knn/indices/ModelCache.java

```diff
@@ -103,7 +103,8 @@ private void initCache() {
      * @return total weight
      */
     public long getTotalWeight() {
-        return cache.asMap().values().stream().map(bytes -> (long) bytes.length).reduce(0L, Long::sum);
+        return cache.asMap().values().stream().map(model -> (long) model.getModelBlob().length)
```

Comment:
nit: since ModelBlob is final, if this method is going to be called often, why not create a method to store length and use it, instead of retrieving blob and calculate length?

Reply:
So model blob is just a byte array and calling length gets the property in the array. I can write a getter to improve readability in Model that would return `modelBlob.length`. However, storing the length as a property of Model I do not think would have any performance benefits.